cargo-dist/templates/installer/installer.sh.j2

```diff
@@ -287,93 +290,84 @@ install() {
     # Potentially-late-bound version of env_script_path to write to rcfiles like $HOME/.profile
     local _env_script_path_expr
 
-{% if install_path.kind == "CargoHome" %}
-    # first try CARGO_HOME, then fallback to HOME
-    if [ -n "${CARGO_HOME:-}" ]; then
-        _install_dir="$CARGO_HOME/bin"
-        _env_script_path="$CARGO_HOME/env"
-
-        # If we need to override the above
-        if [ -n "${CARGO_DIST_FORCE_INSTALL_DIR:-}" ] && [ "${CARGO_DIST_FORCE_INSTALL_DIR}" != "$CARGO_HOME" ]; then
-            _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR/bin"
-            _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
-        fi
-
-        # If CARGO_HOME was set but it ended up being the default $HOME-based path,
-        # then keep things late-bound. Otherwise bake the value for safety.
-        # This is what rustup does, and accurately reproducing it is useful.
-        if [ -n "${HOME:-}" ]; then
-            if [ "$HOME/.cargo/bin" = "$_install_dir" ]; then
+    # Before actually consulting the configured install strategy, see
+    # if we're overriding it.
+    if [ -n "${CARGO_DIST_FORCE_INSTALL_DIR:-}" ]; then
+        {% if install_paths| selectattr("kind", "equalto", "CargoHome") %}
+        _install_dir="$CARGO_DIST_FORCE_INSTALL_DIR/bin"
+        _env_script_path="$CARGO_DIST_FORCE_INSTALL_DIR/env"
+        _install_dir_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/bin" | sed "s,$HOME,\$HOME,")"
+        _env_script_path_expr="$(echo "$CARGO_DIST_FORCE_INSTALL_DIR/env" | sed "s,$HOME,\$HOME,")"
```

Comment:
ooh i just realized that for all our paranoid checking of `$HOME` existing, these expressions will be messed up if it's not set (right..?)

Reply:
It sure will! `sed: first RE may not be empty`

Reply:
That's why I originally still had the more explicit `$HOME` check here. I should restore it.